.github/workflows/build.yml

```diff
@@ -62,37 +81,79 @@ jobs:
           else
             bumpver update --tag alpha --tag-num
           fi
+          echo "TARGET_ENV=SIT" >> $GITHUB_ENV
       - name: Bump rc version
-        if: startsWith(github.ref, 'refs/heads/release/')
-        run: bumpver --tag rc --tag-num
+        if: startsWith(github.ref, 'refs/heads/release/') && github.event_name != 'workflow_dispatch'
+        run: |
+          bumpver --tag rc --tag-num
+          echo "TARGET_ENV=UAT" >> $GITHUB_ENV
       - name: Release version
-        if: github.ref == 'refs/heads/main'
-        run: bumpver --tag final
-
+        if: github.ref == 'refs/heads/main' && github.event_name != 'workflow_dispatch'
+        run: |
+          bumpver --tag final
       #   -- Build --
       - name: Build lambda package
-        run: ./build.sh
+        id: package-build
+        run: |
+          ./build.sh
+          echo "deploy_env=${{ env.TARGET_ENV }}" >> $GITHUB_OUTPUT
       - name: Upload packaged zip
         uses: actions/upload-artifact@v3.1.0
         with:
           name: dist
           path: dist/*.zip
 
-      #   -- Terraform --
-      - name: Deploy to SIT
-        if: github.ref == 'refs/heads/develop'
+  deploy:
+    name: Deploy
+    needs: build
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    environment: 
+      name: ${{ needs.build.outputs.deploy_env }}
+    if: |
+      github.ref == 'refs/heads/develop' ||
+      github.ref == 'refs/heads/main'    ||
+      startsWith(github.ref, 'refs/heads/release') ||
+      github.event_name == 'workflow_dispatch'
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: us-west-2
+          role-session-name: GitHubActions
+          aws-access-key-id: ${{ secrets[vars.AWS_ACCESS_KEY_ID_SECRET_NAME] }}
+          aws-secret-access-key: ${{ secrets[vars.AWS_SECRET_ACCESS_KEY_SECRET_NAME] }}
+          mask-aws-account-id: true
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{ github.repository }}
```

Comment:
Is it required to specify this? I thought by default checkout will checkout the current repo the workflow is being run in

Reply:
That's just sort of the standard way it's always shown, but yeah basically this is explicitly setting the default value.  We can remove it across all repos if you like